pkgs/tools/typesetting/biber/versions.nix

```diff
@@ -0,0 +1,30 @@
+{ lib, callPackage, fetchFromGitHub, fetchpatch, biber }:
+
+{
+  biber-from-texlive = biber;
+  biber-for-tectonic = let version = "2.17"; in (
+    biber.override {
+      ## this is necessary, otherwise wrong `meta.name`
+      texlive.pkgs.biber.texsource = {
+        inherit version;
+        inherit (biber) pname meta;
+      };
+    }
```

Comment:
It is necessary to first override the `version` data here, which is passed to `buildPerlModule`, and then passed to `mkDerivation`. If we simply do `biber.overrideAttrs` the resulting package name will be wrong (`perl5.38.0-biber-2.19` instead of `2.17`), since it has already been preprocessed by `buildPerlModule`.